services/src/main/java/org/keycloak/broker/saml/mappers/AttributeToRoleMapper.java

```diff
@@ -115,19 +117,31 @@ public void importNewUser(KeycloakSession session, RealmModel realm, UserModel u
             user.grantRole(role);
         }
     }
+    
+    /**
+     * Check whether there is a match between an Attribute name, friendly-name and/or value. 
+     * The value must be of String type; corresponding to the {@linkplain SAMLParserUtil#parseAttributeValue} xsi-type mapping.
+     * 
+     * @param mapperModel mapping to evaluate
+     * @param context context with Assertion
+     * @return true if present
+     */
 
     protected boolean isAttributePresent(IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        // match against optional Attribute attributes 'name' and 'friendlyName' and values.
+        // name/friendly-name filters are inactive if configured as null, empty or whitespace
         String name = mapperModel.getConfig().get(ATTRIBUTE_NAME);
-        if (name != null && name.trim().equals("")) name = null;
+        if (!StringUtil.isNotNull(name)) name = null;
         String friendly = mapperModel.getConfig().get(ATTRIBUTE_FRIENDLY_NAME);
-        if (friendly != null && friendly.trim().equals("")) friendly = null;
+        if (!StringUtil.isNotNull(friendly)) friendly = null;
         String desiredValue = mapperModel.getConfig().get(ATTRIBUTE_VALUE);
         AssertionType assertion = (AssertionType)context.getContextData().get(SAMLEndpoint.SAML_ASSERTION);
         for (AttributeStatementType statement : assertion.getAttributeStatements()) {
             for (AttributeStatementType.ASTChoiceType choice : statement.getAttributes()) {
                 AttributeType attr = choice.getAttribute();
-                if (name != null && !name.equals(attr.getName())) continue;
-                if (friendly != null && !name.equals(attr.getFriendlyName())) continue;
+                if (name != null && !Objects.areEqual(name, attr.getName())) continue;
```

Comment:
Keep the amount of changes minimal. Update the original condition if necessary, use Objects.equals instead of areEqual. Use e.g. 
```java
if (name != null && ! Objects.equals(name, attr.getName())) continue;
```